core/templates/dev/head/pages/exploration_editor/editor_tab/StateHints.js

```diff
@@ -51,6 +51,13 @@ oppia.controller('StateHints', [
     };
 
     $scope.changeActiveHintIndex = function(newIndex) {
+      var currentActiveIndex = $scope.activeHintIndex;
+      if (currentActiveIndex && (
+        !stateHintsService.displayed[currentActiveIndex].hintText)) {
+        alertsService.addInfoMessage('Deleting empty hint!');
```

Comment:
I think I'd drop the exclamation mark here, and follow the style of the other info messages -- deleting a hint is not _that_ exciting :P